_rules/explicit-SVG-image-has-name-7d6734.md

```diff
@@ -0,0 +1,179 @@
+
+## Expectation
+
+Each target element has an [accessible name](#accessible-name) that is not only [whitespace](#whitespace).
+
+## Assumptions
+
+This rule assumes that the presence of one of the roles outlined in the applicability indicates the authors intent to include the element in the accessibility tree and thus convey information to the user about that element.
+
+## Accessibility Support
+
+The [HTML Accessibility API Mappings](https://www.w3.org/TR/html-aam-1.0/#html-element-role-mappings) specify that the `<svg>` element has an implicit role of `graphics-document`. However browser support for the `graphics-document` role and the [SVG Accessibility API Mappings](https://www.w3.org/TR/svg-aam-1.0) is inconsistent.
+
+This rule is limited to the explicit use of roles, as a clear indication that content should convey meaning, until the [SVG Accessibility API Mappings](https://www.w3.org/TR/svg-aam-1.0) is more stable and browser support is more consistent.
+
+Browser and assistive technology support for SVG `<title>` and `<desc>` elements is currently inconsistent. Using WAI ARIA in combination with the `img` role for non-decorative `<svg>` elements significantly improves accessibility browser support.
+
+Until browser support for the [SVG Accessibility API Mappings](https://www.w3.org/TR/svg-aam-1.0) is more consistent it is recommended to explicitly remove decorative <svg> elements from the accessibility tree.
+
+## Background
+
+- [Understanding Success Criterion 1.1.1: Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
+- [SVG Accessibility API Mappings (working draft)](https://www.w3.org/TR/svg-aam-1.0)
+- [WAI-ARIA Graphics Module](https://www.w3.org/TR/graphics-aria-1.0/)
+- [ARIA4: Using a WAI-ARIA role to expose the role of a user interface component](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA4)
+- [ARIA6: Using aria-label to provide labels for objects](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA6)
+- [ARIA10: Using aria-labelledby to provide a text alternative for non-text content](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA10)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The `svg` element has an explicit role of `img` and an accessible name from the `title` element that does not contain only whitespace.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<title>1 circle</title>
+	<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Passed Example 2
+
+The `circle` element has an explicit role of `graphics-symbol` and an accessible name from the `aria-label` attribute that does not contain only whitespace.
+
+```html
+<p>How many circles are there?</p>
+<svg xmlns="https://www.w3.org/2000/svg">
```

Comment:
What w3c spec is being used to define the accessible name calculation for this SVG? From my reading of the note in:
https://www.w3.org/TR/svg-aam-1.0/#mapping_additional_nd
There are five ways too find the accessible name of an SVG:
1. aria-labelledby
2. aria-label
3. a direct child title element
4. xlink:title attribute on a link
5. for text container elements, the text content
Unless there is something subtle I'm missing about a circle element with role=graphics-symbol, that doesn't count as a text container and none of the other methods apply, so the SVG does not have an accessible name.

Reply:
The `svg` element doesn't. But given that it also doesn't have one of the required explicit roles, it is inapplicable for this rule.
The `circle` element has one of the required explicit roles, thus is applicable; and has an accessible name given by the `aria-label`, thus it passes the rule.
The example description is about why the `circle` passes the rule, and says nothing about the `svg`.

Reply:
Ahh, that makes more sense. So we are not testing the outer SVG element but just the inner circle element. I guess when I read the rule title saying an "…SVG image has accessible name", I thought all the tests were for the SVG element. Many of the test cases are for the SVG elements. Maybe this should be split up for SVG elements and children of SVG elements. Anyway, thanks for clarifying.